src/expr/src/explain/mod.rs

```diff
@@ -32,6 +33,7 @@ pub struct ExplainContext<'a> {
     pub humanizer: &'a dyn ExprHumanizer,
     pub used_indexes: UsedIndexes,
     pub finishing: Option<RowSetFinishing>,
+    pub duration: Option<Duration>,
```

Comment:
We always have a `Some` value after an explain, right? Would it make sense to not make this an `Option` and initialize to `Duration::ZERO` instead? Saves you some unwrapping later on.

Reply:
> We always have a Some value after an explain, right? 

No, we only have it if you write

```rust
EXPLAIN WITH(timing)
```

Without this option the explain output should not include the one-line footer with the duration value.

Reply:
Yes, but you still have a `Some` value in this field, no? In `drain_explainable_entries` we always update the duration without checking the option. We just decide not to print it if the option is not set:
```
        if self.context.config.timing {
             if let Some(duration) = self.context.duration {
                 writeln!(f, "")?;
                 writeln!(f, "Optimization time: {duration:?}")?;
             }
         }
```

If this wasn't an option, this could be simplified to:
```
        if self.context.config.timing {
             writeln!(f, "")?;
             writeln!(f, "Optimization time: {:?}", self.context.duration)?;
         }
```

Reply:
Ah, yes.